url3aa01d34ec_katsuster_ememu_tgz-pJ8-net_katsuster_ememu_ui_MainJ8/src/net/katsuster/ememu/arm/core/ExecStageARMv5.java

```diff
@@ -0,0 +1,4246 @@
+     */
+    public void executeALUAnd(InstructionARM inst, boolean exec) {
+        int id = inst.getOpcodeSBitShiftID();
+        boolean s = inst.getSBit();
+        int rn = inst.getRnField();
+        int rd = inst.getRdField();
+        int opr = getAddrMode1(inst);
+        int left, right, dest;
+
+        if (!exec) {
+            printALUDisasm(inst, id);
+            return;
+        }
+
+        if (!inst.satisfiesCond(getCPSR())) {
+            return;
+        }
+
+        left = getReg(rn);
+        right = opr;
+        dest = left & right;
+
+        if (s && rd == 15) {
+            getCPSR().setValue(getSPSR());
+        } else if (s) {
+            getCPSR().setNBit(BitOp.getBit32(dest, 31));
+            getCPSR().setZBit(dest == 0);
+            getCPSR().setCBit(getAddrMode1Carry(inst));
+            //V flag is unaffected
+        }
+
+        setReg(rd, dest);
+    }
+
+    /**
+     * 排他的論理和命令。
+     *
+     * @param inst ARM 命令
+     * @param exec デコードと実行なら true、デコードのみなら false
+     */
+    public void executeALUEor(InstructionARM inst, boolean exec) {
+        int id = inst.getOpcodeSBitShiftID();
+        boolean s = inst.getSBit();
+        int rn = inst.getRnField();
+        int rd = inst.getRdField();
+        int opr = getAddrMode1(inst);
+        int left, right, dest;
+
+        if (!exec) {
+            printALUDisasm(inst, id);
+            return;
+        }
+
+        if (!inst.satisfiesCond(getCPSR())) {
+            return;
+        }
+
+        left = getReg(rn);
+        right = opr;
+        dest = left ^ right;
+
+        if (s && rd == 15) {
+            getCPSR().setValue(getSPSR());
+        } else if (s) {
+            getCPSR().setNBit(BitOp.getBit32(dest, 31));
+            getCPSR().setZBit(dest == 0);
+            getCPSR().setCBit(getAddrMode1Carry(inst));
+            //V flag is unaffected
+        }
+
+        setReg(rd, dest);
+    }
+
+    /**
+     * 減算命令。
+     *
+     * @param inst ARM 命令
+     * @param exec デコードと実行なら true、デコードのみなら false
+     */
+    public void executeALUSub(InstructionARM inst, boolean exec) {
+        int id = inst.getOpcodeSBitShiftID();
+        boolean s = inst.getSBit();
+        int rn = inst.getRnField();
+        int rd = inst.getRdField();
+        int opr = getAddrMode1(inst);
+        int left, right, dest;
+
+        if (!exec) {
+            printALUDisasm(inst, id);
+            return;
+        }
+
+        if (!inst.satisfiesCond(getCPSR())) {
+            return;
+        }
+
+        left = getReg(rn);
+        right = opr;
+        dest = left - right;
+
+        if (s && rd == 15) {
+            getCPSR().setValue(getSPSR());
+        } else if (s) {
+            getCPSR().setNBit(BitOp.getBit32(dest, 31));
+            getCPSR().setZBit(dest == 0);
+            getCPSR().setCBit(!IntegerExt.borrowFrom(left, right));
+            getCPSR().setVBit(IntegerExt.overflowFrom(left, right, false));
+        }
+
+        setReg(rd, dest);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 1305:1334, 1344:1373.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.